sdk_experimental/lib/opentelemetry/sdk/trace/propagation/trace_context/response_text_map_propagator.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Trace
+    module Propagation
+      module TraceContext
+        # Propagates trace response using the W3C Trace Context format
+        # https://w3c.github.io/trace-context/#traceresponse-header
+        class ResponseTextMapPropagator
+          TRACERESPONSE_KEY = 'traceresponse'
+          FIELDS = [TRACERESPONSE_KEY].freeze
+
+          private_constant :TRACERESPONSE_KEY, :FIELDS
+
+          # Inject trace context into the supplied carrier.
+          #
+          # @param [Carrier] carrier The mutable carrier to inject trace context into
+          # @param [Context] context The context to read trace context from
+          # @param [optional Setter] setter If the optional setter is provided, it
+          #   will be used to write context into the carrier, otherwise the default
+          #   text map setter will be used.
+          def inject(carrier, context: Context.current, setter: Context::Propagation.text_map_setter)
+            span_context = Trace.current_span(context).context
+            return unless span_context.valid?
+
+            setter.set(carrier, TRACERESPONSE_KEY, TraceParent.from_span_context(span_context).to_s)
+            nil
```

Comment:
Do you need to return the `carrier` here? The propagators [elsewhere](https://github.com/open-telemetry/opentelemetry-ruby/blob/32f4a7a10ecce597a574a675fa797e29561c473d/propagator/jaeger/lib/opentelemetry/propagator/jaeger/text_map_propagator.rb#L81) do, but I'm not sure if it needs to be done here.

Reply:
I used the [Trace Context TextMapPropagator][1] as a reference, which returns nil, but I don't have a preference either way, whatever y'all think is better!

[1]: https://github.com/open-telemetry/opentelemetry-ruby/blob/32f4a7a10ecce597a574a675fa797e29561c473d/api/lib/opentelemetry/trace/propagation/trace_context/text_map_propagator.rb#L32

Reply:
It should return `nil` . The core set of TextMapPropagators were careful about this, e.g. https://github.com/open-telemetry/opentelemetry-ruby/blob/32f4a7a10ecce597a574a675fa797e29561c473d/api/lib/opentelemetry/context/propagation/text_map_propagator.rb#L39

Others, like the Jaeger propagator, should be fixed.

Reply:
That would be the answer, then. 😄